scylla/src/client/session_test.rs

```diff
@@ -578,57 +576,155 @@ async fn test_token_calculation() {
-    session
-        .ddl(format!(
-            "CREATE TABLE IF NOT EXISTS {}.t3 (a text primary key)",
-            ks
-        ))
-        .await
-        .unwrap();
-
-    // Refresh metadata as `ClusterState::compute_token` use them
-    session.await_schema_agreement().await.unwrap();
-    session.refresh_metadata().await.unwrap();
+    session.use_keyspace(ks.as_str(), true).await.unwrap();
 
-    let prepared_statement = session
-        .prepare(format!("INSERT INTO {}.t3 (a) VALUES (?)", ks))
-        .await
-        .unwrap();
-
-    // Try calculating tokens for different sizes of the key
-    for i in 1..50usize {
-        eprintln!("Trying key size {}", i);
-        let mut s = String::new();
-        for _ in 0..i {
-            s.push('a');
-        }
-        let values = (&s,);
+    #[allow(clippy::too_many_arguments)]
+    async fn assert_tokens_equal(
+        session: &Session,
+        prepared: &PreparedStatement,
+        all_values_in_query_order: impl SerializeRow,
+        token_select: &PreparedStatement,
+        pk_ck_values: impl SerializeRow,
+        ks_name: &str,
+        table_name: &str,
+        pk_values: impl SerializeRow,
+    ) {
         session
-            .execute_unpaged(&prepared_statement, &values)
+            .execute_unpaged(prepared, &all_values_in_query_order)
             .await
             .unwrap();
 
         let (value,): (i64,) = session
-            .query_unpaged(
-                format!("SELECT token(a) FROM {}.t3 WHERE a = ?", ks),
-                &values,
-            )
+            .execute_unpaged(token_select, &pk_ck_values)
             .await
             .unwrap()
             .into_rows_result()
             .unwrap()
             .single_row::<(i64,)>()
             .unwrap();
         let token = Token::new(value);
-        let prepared_token = Murmur3Partitioner
-            .hash_one(&prepared_statement.compute_partition_key(&values).unwrap());
+        let prepared_token = prepared
+            .calculate_token(&all_values_in_query_order)
+            .unwrap()
+            .unwrap();
         assert_eq!(token, prepared_token);
         let cluster_state_token = session
             .get_cluster_state()
-            .compute_token(&ks, "t3", &values)
+            .compute_token(ks_name, table_name, &pk_values)
             .unwrap();
         assert_eq!(token, cluster_state_token);
     }
+
+    // Different sizes of the key
+    {
+        session
+            .ddl("CREATE TABLE IF NOT EXISTS t1 (a text primary key)")
+            .await
+            .unwrap();
+
+        let prepared_statement = session
+            .prepare("INSERT INTO t1 (a) VALUES (?)")
+            .await
+            .unwrap();
+
+        let token_selection = session
+            .prepare("SELECT token(a) FROM t1 WHERE a = ?")
+            .await
+            .unwrap();
+
+        for i in 1..50usize {
+            eprintln!("Trying key size {}", i);
```

Comment:
:question: :pick: Why not `tracing::debug!`?

Reply:
Idk, that's how it was before.